extension/encoding/googlecloudlogentryencodingextension/PERFORMANCE_ANALYSIS.md

```diff
@@ -0,0 +1,169 @@
+# Performance Analysis: UnmarshalLogs Implementation
+
+## Current Implementation Issues
+
+### 1. Structure Overhead
+The current `UnmarshalLogs` implementation creates a **new ResourceLogs and ScopeLogs for every single log line**:
+
+```go
+func (ex *ext) handleLogLine(logs plog.Logs, logLine []byte) error {
+    // ...
+    rl := logs.ResourceLogs().AppendEmpty()  // NEW ResourceLogs for each log
+    r := rl.Resource()
+    scopeLogs := rl.ScopeLogs().AppendEmpty()  // NEW ScopeLogs for each log
+    // ...
+}
+```
+
+**Performance Impact:**
+- For N log lines, this creates N ResourceLogs and N ScopeLogs structures
+- Each ResourceLogs contains:
+  - Resource attributes map
+  - Schema URL
+  - ScopeLogs slice
+- Each ScopeLogs contains:
+  - InstrumentationScope (name, version, attributes)
+  - Schema URL  
+  - LogRecords slice
+
+**Memory Overhead:** For 1000 log lines, this creates:
+- 1000 ResourceLogs structures
+- 1000 ScopeLogs structures
+- Significant overhead in slice allocations and memory fragmentation
```

Comment:
Hm... This was something I discussed in the beginning, and decided to go with just Resource logs (we do group them in the AWS logs encoding, for example). Can you actually prove that this is worse using a benchmark?

So, what performs better:
1. Use a resource log for every log line
2. Group by resource
   - BIG issue: the labels should be resource attributes. How are you going to make this perform well in the code (doesn't degrade CPU or memory)?

If we can't have this as a definite conclusion, then this should not be merged or shared (I know this is only a draft, but I am doubting these conclusions).

Thanks for bringing this up!